### PR TITLE
fix: insert and delete executor

### PR DESCRIPTION
**Pull request by aryan-rajoria**
Fixing issues in insert and delete executor based on feedback from @gaurav274 

- [x] Using list comprehension in `insert_executor`.
- [x] Removing `_roll_back` from the executors.
- [x] Using descriptive variable names.
- [x] Applying predicate function in `delete_executor` using the SQLAlchemy.
- [x] Move `single_predicate_to_filter()` to ExpressionUtils.
- [x] Make it more robust (take into consideration `AND` and `OR`)
- [x] Adding test cases for `single_predicate()`.

### CODE REVIEW

**Comment by jarulraj** [comment]
@aryan-rajoria `single_predicate_node_to_filter` is missing.

**Comment by jarulraj** [comment]
@aryan-rajoria I brought the PR up-to-date with master. Please pull in the latest changes.

**Comment by jarulraj** [comment]
This delete statement does not work -- """DELETE FROM testDeleteOne WHERE id < 20  AND dummyfloat < 2;"""

**Comment by jarulraj** [comment]
> This delete statement does not work -- """DELETE FROM testDeleteOne WHERE id < 20 AND dummyfloat < 2;"""

This is now resolved. All the tasks have been completed.

File: eva/expression/expression_utils.py [hunk 1]
```diff
@@ -297,3 +300,51 @@ def _has_simple_expressions(expr):
...
+        elif predicate_node.etype == ExpressionType.COMPARE_GREATER:
+            filter_clause = x > y
+        elif predicate_node.etype == ExpressionType.COMPARE_LESSER:
+            filter_clause = x < y
+        elif predicate_node.etype == ExpressionType.COMPARE_GEQ:
+            filter_clause = x >= y
+        elif predicate_node.etype == ExpressionType.COMPARE_LEQ:
+            filter_clause = x <= y
+        elif predicate_node.etype == ExpressionType.COMPARE_NEQ:
+            filter_clause = x != y
```
Comment (xzdandy) [inline eva/expression/expression_utils.py:348]: `CONTAINS` and `IS_CONTAINED` are not supported here. We can check and raise exception if they are non-trivial.

File: eva/storage/sqlite_storage_engine.py [hunk 2]
```diff
@@ -196,22 +198,14 @@ def delete(self, table: TableCatalogEntry, where_clause: Dict[str, Any]):
...
-            table_columns = [
-                col.name
-                for col in table_to_delete_from.columns
-                if col.name != "_row_id"
-            ]
-            filter_clause = []
+
+            filter_clause = predicate_node_to_filter_clause(
+                table=table_to_delete_from, predicate_node=where_clause
+            )
```
Comment (xzdandy) [inline eva/storage/sqlite_storage_engine.py:204]: Nit: I feel it is better to move `predicate_node_to_filter_clause` logic into `delete_executor` and the `delete_executor` accepts the `ColumnElement[bool]` as the input argument (i.e., where_clause).  `ColumnElement[bool]` is the return type of `and_` and `or_`

File: eva/expression/expression_utils.py (hunk 1 above)
Comment (jarulraj) [reply eva/expression/expression_utils.py]: Resolved.

File: eva/storage/sqlite_storage_engine.py (hunk 2 above)
Comment (jarulraj) [reply eva/storage/sqlite_storage_engine.py]: Good idea. Moved it to `delete_executor`

**Review by gaurav274 — APPROVED** [review]

**Pull request merged**